lighthouse-core/gather/drivers/driver.js

```diff
@@ -219,6 +210,28 @@ class Driver {
   }
 
   /**
+   * If our main document URL redirects, we will update options.url accordingly
+   * As such, options.url will always represent the post-redirected URL.
+   * options.initialUrl is the pre-redirect URL that things started with
+   *
+   * Caveat: only works when network recording enabled for a pass
```

Comment:
should we file a bug for this? I'm thinking either network recording should always happen, getting rid of `config.network`, or network recording should always happen but the results are discarded if `config.network` is `false`